util/logger/src/lib.rs

```diff
@@ -174,3 +177,32 @@ pub fn init(config: Config) -> Result<LoggerInitGuard, SetLoggerError> {
 pub fn flush() {
     log::logger().flush()
 }
+
+fn setup_panic_logger() {
+    // let next = panic::take_hook();
+    let panic_logger = |info: &panic::PanicInfo| {
+        let backtrace = Backtrace::new();
+        let thread = thread::current();
+        let name = thread.name().unwrap_or("unnamed");
+        let location = info.location().unwrap(); // The current implementation always returns Some
+        let msg = match info.payload().downcast_ref::<&'static str>() {
+            Some(s) => *s,
+            None => match info.payload().downcast_ref::<String>() {
+                Some(s) => &**s,
+                None => "Box<Any>",
+            },
+        };
+        log::error!(
+            target: "panic", "thread '{}' panicked at '{}': {}:{}{:?}",
+            name,
+            msg,
+            location.file(),
+            location.line(),
+            backtrace,
+        );
+    };
+    panic::set_hook(Box::new(move |info| {
```

Comment:
This conflicts with sentry:

> Registers a custom panic hook, replacing any that was previously registered.

Reply:
https://github.com/getsentry/sentry-rust/blob/bdffebe356212dab7968e72941bed77a56556e59/src/integrations/panic.rs#L84

Reply:
> This conflicts with sentry:
> 
> > Registers a custom panic hook, replacing any that was previously registered.

It says dispatches (take the current hook and call after report sent)
``` rust
pub fn register_panic_handler() {
    let next = panic::take_hook();
    panic::set_hook(Box::new(move |info| {
        panic_handler(info);
        next(info);
    }));
}
```

Reply:
Maybe we can do the same as `[sentry](https://github.com/getsentry/sentry-rust/blob/bdffebe356212dab7968e72941bed77a56556e59/src/integrations/panic.rs#L84)` do, so we do not care about the order.

Reply:
> Maybe we can do the same as `[sentry](https://github.com/getsentry/sentry-rust/blob/bdffebe356212dab7968e72941bed77a56556e59/src/integrations/panic.rs#L84)` do, so we do not care about the order.

Yes, I am considering about it. I added a comment [here](https://github.com/nervosnetwork/ckb/pull/559/files#diff-15ed784c2cdfaaa8f2b56f024da15f55R206), the disadvantage is that stdout will display twice panic info since logger's output redirect to stdout as well.

Reply:
@TheWaWaR @doitian I decide to follow @TheWaWaR 's addvice to wrap previous hook inside panic_logger. It is a safer way.

As for displaying twice in stdout, I think, that is ok, just ignore it.